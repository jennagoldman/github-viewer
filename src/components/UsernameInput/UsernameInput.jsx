import React from 'react';
import PropTypes from 'prop-types';

const UsernameInput = ({ username, onUsernameChange, onSearch }) => (
  <>
    <form onSubmit={onSearch}>
      <input type="text" name="username" value={username} onChange={onUsernameChange}/>
      <button>Lookup User</button>
    </form>
  </>
);

UsernameInput.propTypes = {
  username: PropTypes.string.isRequired,
  onUsernameChange: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired
};

export default UsernameInput;
