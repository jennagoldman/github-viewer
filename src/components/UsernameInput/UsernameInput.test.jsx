import React from 'react';
import { shallow } from 'enzyme';
import UsernameInput from './UsernameInput.jsx';

describe('UsernameInput', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<UsernameInput username="jennagoldman" onUsernameChange={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
