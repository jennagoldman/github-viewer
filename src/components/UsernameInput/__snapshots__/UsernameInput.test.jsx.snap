// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernameInput matches a snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      name="username"
      onChange={[Function]}
      type="text"
      value="jennagoldman"
    />
    <button>
      Lookup User
    </button>
  </form>
</Fragment>
`;
